### DISCUSSION TITLE
No model found and No valid best SED found

### DISCUSSION THREAD

**Discussion by franklynda** [mboquien/cigale#32] [Q&A]
I run various data files using latest version of CIGALE 2022  but no model was found and when I did pcigale-plots sed it gives no valid sed found. Please help me out of this situation. The screenshot and the pcigale init  are attached.

# File containing the input data. The columns are 'id' (name of the
# object), 'redshift' (if 0 the distance is assumed to be 10 pc),
# 'distance' (Mpc, optional, if present it will be used in lieu of the
# distance computed from the redshift), the filter names for the fluxes,
# and the filter names with the '_err' suffix for the uncertainties. The
# fluxes and the uncertainties must be in mJy for broadband data and in
# W/m² for emission lines. This file is optional to generate the
# configuration file, in particular for the savefluxes module.
data_file = my data

# Optional file containing the list of physical parameters. Each column
# must be in the form module_name.parameter_name, with each line being a
# different model. The columns must be in the order the modules will be
# called. The redshift column must be the last one. Finally, if this
# parameter is not empty, cigale will not interpret the configuration
# parameters given in pcigale.ini. They will be given only for
# information. Note that this module should only be used in conjonction
# with the savefluxes module. Using it with the pdf_analysis module will
# yield incorrect results.
parameters_file = 

# Available modules to compute the models. The order must be kept.
# SFH:
# * sfh2exp (double exponential)
# * sfhdelayed (delayed SFH with optional exponential burst)
# * sfhdelayedbq (delayed SFH with optional constant burst/quench)
# * sfhfromfile (arbitrary SFH read from an input file)
# * sfhperiodic (periodic SFH, exponential, rectangle or delayed)
# SSP:
# * bc03 (Bruzual and Charlot 2003)
# * m2005 (Maraston 2005; note that it cannot be combined with the nebular module)
# Nebular emission:
# * nebular (continuum and line nebular emission)
# Dust attenuation:
# * dustatt_modified_CF00 (modified Charlot & Fall 2000 attenuation law)
# * dustatt_modified_starburst (modified Calzetti 2000 attenuaton law)
# Dust emission:
# * casey2012 (Casey 2012 dust emission models)
# * dale2014 (Dale et al. 2014 dust emission templates)
# * dl2007 (Draine & Li 2007 dust emission models)
# * dl2014 (Draine et al. 2014 update of the previous models)
# * themis (Themis dust emission models from Jones et al. 2017)
# AGN:
# * skirtor2016 (AGN models from Stalevski et al. 2012, 2016)
# * fritz2006 (AGN models from Fritz et al. 2006)
# X-ray:
# * xray (from AGN and galaxies; skirtor2016/fritz2006 is needed for AGN)
# Radio:
# * radio (galaxy synchrotron emission and AGN; skirtor2016/fritz2006 is needed for AGN)
# Restframe parameters:
# * restframe_parameters (UV slope (β), IRX, D4000, EW, etc.)
# Redshift+IGM:
# * redshifting (mandatory, also includes the IGM from Meiksin 2006)
sed_modules = sfhdelayed, bc03, nebular, dustatt_modified_CF00, dale2014, redshifting

# Method used for statistical analysis. Available methods: pdf_analysis,
# savefluxes.
analysis_method = pdf_analysis

# Number of CPU cores available. This computer has 4 cores.
cores = 4

# Bands to consider. To consider uncertainties too, the name of the band
# must be indicated with the _err suffix. For instance: FUV, FUV_err.
bands = mag_i, mag_u, mag_r, mag_g, mag_g_err, mag_u_err, mag_r_err, mag_i_err

# Properties to be considered. All properties are to be given in the
# rest frame rather than the observed frame. This is the case for
# instance the equivalent widths and for luminosity densities.
properties = 

# Configuration of the SED creation modules.
[sed_modules_params]
  
  [[sfhdelayed]]
    # e-folding time of the main stellar population model in Myr.
    tau_main = 2000.0
    # Age of the main stellar population in the galaxy in Myr. The precision
    # is 1 Myr.
    age_main = 5000
    # e-folding time of the late starburst population model in Myr.
    tau_burst = 50.0
    # Age of the late burst in Myr. The precision is 1 Myr.
    age_burst = 20
    # Mass fraction of the late burst population.
    f_burst = 0.0
    # Multiplicative factor controlling the SFR if normalise is False. For
    # instance without any burst: SFR(t)=sfr_A×t×exp(-t/τ)/τ²
    sfr_A = 1.0
    # Normalise the SFH to produce one solar mass.
    normalise = True
  
  [[bc03]]
    # Initial mass function: 0 (Salpeter) or 1 (Chabrier).
    imf = 0
    # Metalicity. Possible values are: 0.0001, 0.0004, 0.004, 0.008, 0.02,
    # 0.05.
    metallicity = 0.02
    # Age [Myr] of the separation between the young and the old star
    # populations. The default value in 10^7 years (10 Myr). Set to 0 not to
    # differentiate ages (only an old population).
    separation_age = 10
  
  [[nebular]]
    # Ionisation parameter
    logU = -2.0
    # Gas metallicity
    zgas = 0.014
    # Electron density
    ne = 100
    # Fraction of Lyman continuum photons escaping the galaxy
    f_esc = 0.0
    # Fraction of Lyman continuum photons absorbed by dust
    f_dust = 0.0
    # Line width in km/s
    lines_width = 300.0
    # Include nebular emission.
    emission = True
  
  [[dustatt_modified_CF00]]
    # V-band attenuation in the interstellar medium.
    Av_ISM = 1.0
    # Av_ISM / (Av_BC+Av_ISM)
    mu = 0.44
    # Power law slope of the attenuation in the ISM.
    slope_ISM = -0.7
    # Power law slope of the attenuation in the birth clouds.
    slope_BC = -1.3
    # Filters for which the attenuation will be computed and added to the
    # SED information dictionary. You can give several filter names
    # separated by a & (don't use commas).
    filters = V_B90 & FUV
  
  [[dale2014]]
    # AGN fraction. It is not recommended to combine this AGN emission with
    # the of Fritz et al. (2006) or SKIRTOR models.
    fracAGN = 0.0
    # Alpha slope. Possible values are: 0.0625, 0.1250, 0.1875, 0.2500,
    # 0.3125, 0.3750, 0.4375, 0.5000, 0.5625, 0.6250, 0.6875, 0.7500,
    # 0.8125, 0.8750, 0.9375, 1.0000, 1.0625, 1.1250, 1.1875, 1.2500,
    # 1.3125, 1.3750, 1.4375, 1.5000, 1.5625, 1.6250, 1.6875, 1.7500,
    # 1.8125, 1.8750, 1.9375, 2.0000, 2.0625, 2.1250, 2.1875, 2.2500,
    # 2.3125, 2.3750, 2.4375, 2.5000, 2.5625, 2.6250, 2.6875, 2.7500,
    # 2.8125, 2.8750, 2.9375, 3.0000, 3.0625, 3.1250, 3.1875, 3.2500,
    # 3.3125, 3.3750, 3.4375, 3.5000, 3.5625, 3.6250, 3.6875, 3.7500,
    # 3.8125, 3.8750, 3.9375, 4.0000
    alpha = 2.0
  
  [[redshifting]]
    # Redshift of the objects. Leave empty to use the redshifts from the
    # input file.
    redshift = 

# Configuration of the statistical analysis method.
[analysis_params]
  # List of the physical properties to estimate. Leave empty to analyse
  # all the physical properties (not recommended when there are many
  # models).
  variables = sfh.sfr, sfh.sfr10Myrs, sfh.sfr100Myrs
  # List of bands for which to estimate the fluxes. Note that this is
  # independent from the fluxes actually fitted to estimate the physical
  # properties.
  bands = mag_i, mag_u, mag_r, mag_g
  # If true, save the best SED for each observation to a file.
  save_best_sed = True
  # Save the raw chi2. It occupies ~15 MB/million models/variable. Allowed
  # values are 'all', 'none', 'properties', and 'fluxes'.
  save_chi2 = all
  # Take into account upper limits. If 'full', the exact computation is
  # done. If 'noscaling', the scaling of the models will not be adjusted
  # but the χ² will include the upper limits adequately. Waiving the
  # adjustment makes the fitting much faster compared to the 'full' option
  # while generally not affecting the results in any substantial manner.
  # This is the recommended option as it achieves a good balance between
  # speed and reliability. Finally, 'none' simply discards bands with
  # upper limits.
  lim_flag = noscaling
  # If true, for each object we create a mock object and analyse them.
  mock_flag = False
  # When redshifts are not given explicitly in the redshifting module,
  # number of decimals to round the observed redshifts to compute the grid
  # of models. To disable rounding give a negative value. Do not round if
  # you use narrow-band filters.
  redshift_decimals = 2
  # Number of blocks to compute the models and analyse the observations.
  # If there is enough memory, we strongly recommend this to be set to 1.
  blocks = 1

The screenshot
[INFO] Sanity check of the analysis results.
[ERROR] No suitable model found for 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 
14, 15. It may be that models are older than the universe or that the χ² are 
very large.
[INFO] 0.0% of the objects have χ²_red~0 and 0.0% χ²_red<0.5.
[INFO] Saving the analysis results.
[INFO] Run completed! 👍
[INFO] End: 2022-11-10/12:40:48
[INFO] Total duration: 0:00:07
(py3) victory@victory-HP-EliteBook-Folio-9470m:~/Documents/cigale-v2022.0$ pcigale-plot sed
bash: pcigale-plot: command not found
(py3) victory@victory-HP-EliteBook-Folio-9470m:~/Documents/cigale-v2022.0$ pcigale-plots sed
🌓  Object 4/15 ━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━  27%  0:00:00 -:--:-- 0.0/s/home/victory/Documents/cigale-v2022.0/pcigale_plots/plot_types/sed/__init__.py:148: UserWarning: Warning: converting a masked element to nan.
  mod_fluxes = np.array(
/home/victory/Documents/cigale-v2022.0/pcigale_plots/plot_types/sed/__init__.py:148: UserWarning: Warning: converting a masked element to nan.
  mod_fluxes = np.array(
/home/victory/Documents/cigale-v2022.0/pcigale_plots/plot_types/sed/__init__.py:148: UserWarning: Warning: converting a masked element to nan.
  mod_fluxes = np.array(
/home/victory/Documents/cigale-v2022.0/pcigale_plots/plot_types/sed/__init__.py:148: UserWarning: Warning: converting a masked element to nan.
  mod_fluxes = np.array(
👍 Object 15/15 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%  0:00:00 0:00:00 13.7/s
[WARNING] No valid best SED found for 2. No plot created.
[WARNING] No valid best SED found for 3. No plot created.
[WARNING] No valid best SED found for 1. No plot created.
[WARNING] No valid best SED found for 4. No plot created.
[WARNING] No valid best SED found for 7. No plot created.
[WARNING] No valid best SED found for 8. No plot created.
[WARNING] No valid best SED found for 6. No plot created.
[WARNING] No valid best SED found for 5. No plot created.
[WARNING] No valid best SED found for 11. No plot created.
[WARNING] No valid best SED found for 10. No plot created.
[WARNING] No valid best SED found for 9. No plot created.
[WARNING] No valid best SED found for 12. No plot created.
[WARNING] No valid best SED found for 14. No plot created.
[WARNING] No valid best SED found for 13. No plot created.
[WARNING] No valid best SED found for 15. No plot created.
[INFO] Done.

**Comment by mboquien** [comment]
Looking at your pcigale.ini, you only have a single model. You need to add more values to the parameters that would to let free. Also make sure that your models are younger than the age of the universe at the redshift of the object. Otherwise CIGALE will discard them as physically impossible.
